framework/integration/test/basic/basic_test.go

```diff
@@ -49,9 +49,27 @@ func Test_Finalizer_Integration_Basic(t *testing.T) {
 		t.Fatal("expected", nil, "got", err)
 	}
 
-	// We directly pass the configmap to UpdateFunc.
+	// We update the configmap with a meanless label to ensure a change in the
+	// ResourceVersion of the ConfigMap.
+	cm.SetLabels(
+		map[string]string{
+			"testlabel": "testlabel",
+		},
+	)
+	_, err = client.UpdateConfigMap(cm)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	// We directly pass the _old_ configmap to UpdateFunc.
+	// This is expected to fail, we want to make sure that we only use the latest
+	// ResourceVersion of an object.
+	operatorkitFramework.UpdateFunc(createdConfigMap, createdConfigMap)
+
+	// We run UpdateFunc multiple times on the old object to check for duplicates.
 	operatorkitFramework.UpdateFunc(createdConfigMap, createdConfigMap)
 
+	// We get the current configmap.
```

Comment:
I like the change in general but I think it deservers better tests. It would be good to reply the scenario we were falling into, which I'm not sure what it was.

Was it two frameworks with the same name creating the same finalizers in parallel?

Reply:
Yeah I reckon it was something like this.